### PR TITLE
Feat/record

### PR DESCRIPTION
**Pull request by sgiraldog**

### CODE REVIEW

File: app/Http/Controllers/RecordController.php [hunk 1]
```diff
@@ -14,19 +14,20 @@ public function create()
...
 
-      if(Auth::user()->getRole() == 1){
-        $record = Record::all();
+      if($user->getRole() == 1){
+        $record = Record::where('user_id', $user->getId())->get();
         $data["records"] = $record;
         $chart = new RecordChart;
         $chart->labels($record->pluck('created_at')->pluck('month')->all());
-        $chart->dataset('Weight by month', 'line', $record->pluck('weight')->all());
+        $chart->dataset('Weight by month', 'line', $record->pluck('weight')->all())->color("rgb(0, 102, 255)");
```
Comment (ljpalaciom) [inline app/Http/Controllers/RecordController.php]: Lang please

**Review by ljpalaciom — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/RecordController.php [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    {
+        return view('record.create');
+    }
+
+    public function list(){
+
+      $user = Auth::user();
+
+      $data = []; //to be sent to the view
+      $data["title"] = "Records";
```
Comment (ljpalaciom) [inline app/Http/Controllers/RecordController.php]: Use lang please

File: app/Http/Controllers/RecordController.php [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
```
Comment (ljpalaciom) [inline app/Http/Controllers/RecordController.php]: Please don't add two spaces.

File: app/Http/Controllers/RecordController.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+          $record = Record::all();
+          $data["records"] = $record;
+          return view('record.list')->with("data",$data);
+      }
+
+    }
+
+    public function sort($order){
+      $data = []; //to be sent to the view
+      $data["title"] = "Records";
```
Comment (ljpalaciom) [inline app/Http/Controllers/RecordController.php]: Lang please

File: app/Http/Controllers/RecordController.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    }
+
+
+
+    public function show($id)
+  {
+      $data = []; //to be sent to the view
+      $record = Record::findOrFail($id);
+
+      $data["title"] = "Record";
```
Comment (ljpalaciom) [inline app/Http/Controllers/RecordController.php]: Lang please

File: app/Record.php [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+  {
+    return $this->attributes['id'];
+  }
+
+  public function setId($id)
+  {
+    $this->attributes['id'] = $id;
+  }
+
+
```
Comment (ljpalaciom) [inline app/Record.php]: Please don't add two spaces.

File: routes/web.php [hunk 7]
```diff
@@ -14,6 +14,13 @@
     return redirect('/login');
 });
 
+Route::get('/record/create', 'RecordController@create')->name("record.create");
+Route::post('/record/save', 'RecordController@save')->name("record.save");
+Route::get('/record/sort/{order}','RecordController@sort')->name("record.sort");
+Route::get('/record/list', 'RecordController@list')->name("record.list");
+Route::get('/record/delete/{id}', 'RecordController@delete')->name("record.delete"); //TODO: Change to Delete (Crashes).
```
Comment (ljpalaciom) [inline routes/web.php]: Put post, to solve the problem you have to change the form with the method delete

**Review by ljpalaciom — CHANGES_REQUESTED** [review]

**Comment by sgiraldog** [comment]
Done

**Review by ljpalaciom — APPROVED** [review]

**Pull request merged**